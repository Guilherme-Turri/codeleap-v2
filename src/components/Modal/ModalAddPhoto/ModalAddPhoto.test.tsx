import React from 'react'
import { ModalAddPhoto } from '.'
import { componentRender } from '../../../render/componentRender'
import { screen, fireEvent } from '@testing-library/react'

describe('<ModalAddPhoto>', () => {

  it('should render on screen properly on success', () => {
    const ModalPhotoAddProps = {
      handleAddPhoto: jest.fn(),
      handleCancel: jest.fn(),
      handleSuccessPhoto: jest.fn(),
      message: 'Message',
      loadPhoto: false
    }
    componentRender(<ModalAddPhoto {...ModalPhotoAddProps} />)

    const successbtn = screen.getByText('OK')
    fireEvent.click(successbtn);
    expect(ModalPhotoAddProps.handleSuccessPhoto).toHaveBeenCalled();
  })

  it('should render on screen properly on LoadPhotoTrue', () => {
    const ModalPhotoAddProps = {
      handleAddPhoto: jest.fn(),
      handleCancel: jest.fn(),
      handleSuccessPhoto: jest.fn(),
      message: 'Message',
      loadPhoto: true
    }
    const { container } = componentRender(<ModalAddPhoto {...ModalPhotoAddProps} />)
    const successbtn = screen.getByTestId('successbtn')

    expect(successbtn).toHaveStyleRule('display', 'none');
    expect(container).toMatchSnapshot();
  })

  it('should render on screen properly when adding a photo and cancel this action', () => {
    const ModalPhotoAddProps = {
      handleAddPhoto: jest.fn(),
      handleCancel: jest.fn(),
      handleSuccessPhoto: jest.fn(),
      message: null,
      loadPhoto: false
    }
    componentRender(<ModalAddPhoto {...ModalPhotoAddProps} />)
    const cancelBtn = screen.getByTestId('cancelbtn')
    fireEvent.click(cancelBtn);
    expect(ModalPhotoAddProps.handleCancel).toHaveBeenCalled();

  })
  it('should render on screen properly when adding a photo', () => {
    const ModalPhotoAddProps = {
      handleAddPhoto: jest.fn(),
      handleCancel: jest.fn(),
      handleSuccessPhoto: jest.fn(),
      message: null,
      loadPhoto: false
    }

    componentRender(<ModalAddPhoto {...ModalPhotoAddProps} />)
    const inputElement = screen.getByTestId('input')
    const file = new File(['test content'], 'test.png', { type: 'image/png' });
    fireEvent.change(inputElement, { target: { files: [file] } });

    const uploadBtn = screen.getByTestId('upload')
    fireEvent.click(uploadBtn)
    expect(ModalPhotoAddProps.handleAddPhoto).toHaveBeenCalled();
  })
})
