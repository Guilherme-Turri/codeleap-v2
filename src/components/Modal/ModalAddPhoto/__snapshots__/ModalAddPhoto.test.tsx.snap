// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalAddPhoto> should render on screen properly on LoadPhotoTrue 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 0.313rem;
}

.c0 .c1 {
  font-size: 1.5rem;
}

.c0 input {
  all: unset;
}

.c4 {
  display: none;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 6rem;
  height: 3rem;
  margin: 0.938rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c5 {
  all: unset;
  background-color: #00CBA9;
  border-radius: 0.313rem;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1 c2"
    >
      Choose a picture to upload
    </p>
    <div
      class="c3"
    >
      <p
        class="c1 c2"
      >
        Message
      </p>
      <div
        class="c4"
        data-testid="successbtn"
      >
        <button
          class="c5"
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;
